core/qt-ble.cpp

```diff
@@ -146,6 +155,9 @@ dc_status_t BLEObject::write(const void *data, size_t size, size_t *actual)
 
 	if (!receivedPackets.isEmpty()) {
 		qDebug() << ".. write HIT with still incoming packets in queue";
+		do {
+			receivedPackets.takeFirst();
+		} while (!receivedPackets.isEmpty());
```

Comment:
Am I missing something? Why not `receivedPackets.clear()`?

Reply:
Because I have never used QLists and we already used takeFirst?

Reply:
Fair enough, could have been that I missed something subtle. Note that `QList` is *not* a linked list as the name would imply, but a "deque" based on a list of arrays of pointers. Qt's data structures are a mess.

Reply:
I basically do Qt (and C++ in general) programming by pattern matching. I  do try to look up the online docs, but QList had a ton of different functions, and the sanely named one (removeAll) did something entirely different, and I never noticed "clear()".

Oh well.